src/core/lib/debug/trace.h

```diff
@@ -111,6 +112,472 @@ class SavedTraceFlags {
   std::map<std::string, bool> values_;
 };
 
+extern TraceFlag api_trace;
+#define GRPC_API_TRACE_LOG(level) \
+  LOG_IF(level, grpc_core::api_trace.enabled()) << "(api) "
```

Comment:
Do we want to automatically log the name of the tracer in every log message?  Usually there's enough info already to tell what the log message is for.

Reply:
I'm flexible. A good handful of uses add that information explicitly every time already, and I think it adds some helpful context, but I can support a YAGNI argument.

Reply:
The case I was thinking of is for things like the tracer argument passed to classes like `RefCounted<>` and `InternallyRefCounted<>`, where we explicitly use a different label for the log messages than the name of the tracer, because a single tracer enables logging for multiple objects.  It's not necessary to log the name of the tracer in those cases.